### PR TITLE
feat(shorebird_cli): add `AndroidSdk`

### PR DESCRIPTION
**Pull request by felangel**
## Description

- feat(shorebird_cli): add `AndroidSdk`

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/android_sdk.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+class AndroidSdk {
+  String? get path {
+    String? androidHomeDir;
+    if (platform.environment.containsKey(kAndroidHome)) {
+      androidHomeDir = platform.environment[kAndroidHome];
+    } else if (platform.environment.containsKey(kAndroidSdkRoot)) {
+      androidHomeDir = platform.environment[kAndroidSdkRoot];
+    } else if (platform.isLinux) {
+      final home = platform.environment['HOME'] ?? '~';
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/android_sdk.dart]: ~ is a shell concept, btw.  So Process won't expand it for you.

File: packages/shorebird_cli/lib/src/android_sdk.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+const kAndroidHome = 'ANDROID_HOME';
+const kAndroidSdkRoot = 'ANDROID_SDK_ROOT';
+
+/// A reference to a [AndroidSdk] instance.
+final androidSdkRef = create(AndroidSdk.new);
+
+/// The [AndroidSdk] instance available in the current zone.
+AndroidSdk get androidSdk => read(androidSdkRef);
+
+class AndroidSdk {
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/android_sdk.dart:12]: Add doc comments to this file

**Review by eseidel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/android_sdk.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  String? get path {
+    String? androidHomeDir;
+    if (platform.environment.containsKey(kAndroidHome)) {
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/android_sdk.dart:15]: For consistency with the other android-related files, we should probably either stick to a cascading if/else (as here) or early returns (as elsewhere)

File: packages/shorebird_cli/lib/src/android_sdk.dart (hunk 2 above)
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/android_sdk.dart:12]: Is this being used anywhere?

Comment (felangel) [reply packages/shorebird_cli/lib/src/android_sdk.dart:12]: Not yet will be in the next PR when i introduce adb

File: packages/shorebird_cli/lib/src/android_sdk.dart (hunk 1 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/android_sdk.dart]: yup I adjusted this, good call 👍

**Pull request merged**